src/wallet/wallet2.cpp

```diff
@@ -13046,6 +13055,77 @@ void wallet2::throw_on_rpc_response_error(const boost::optional<std::string> &st
   THROW_WALLET_EXCEPTION_IF(*status != CORE_RPC_STATUS_OK, tools::error::wallet_generic_rpc_error, method, m_trusted_daemon ? *status : "daemon error");
 }
 //----------------------------------------------------------------------------------------------------
+
+bool wallet2::save_to_file(const std::string& path_to_file, const std::string& raw, bool is_hex) const
+{
+  if (is_hex || m_export_format == ExportFormat::Raw)
+  {
+    return epee::file_io_utils::save_string_to_file(path_to_file, raw);
+  }
+
+  std::string h = HEX_OUTPUT_MAGIC + "\n";
+  h.append(epee::string_tools::buff_to_hex_nodelimer(raw));
+
+  // Split the long string into multiple lines, similarly to how openssl
+  // prints certificates.
+  for (size_t i = 63; i < h.length(); i += 64)
+  {
+    h.insert(i, 1, '\n');
+  }
```

Comment:
For very small files, this sounds like a buffer overflow (unless insert just does nothing, I've not looked at its implementation, but in any case an apparent bug.)